src/main/java/io/javalin/core/util/MethodNotAllowedUtil.kt

```diff
@@ -0,0 +1,47 @@
+package io.javalin.core.util
+
+import io.javalin.core.HandlerType
+import io.javalin.core.PathMatcher
+import io.javalin.translator.json.JavalinJsonPlugin.objectToJsonMapper
+
+private data class JsonMethodNotAllowed(val availableMethods: List<HandlerType>)
+
+object MethodNotAllowedUtil {
+
+    @JvmStatic
+    fun findAvailableHandlerTypes(matcher: PathMatcher, requestUri: String): List<HandlerType> {
+        val availableHandlerTypes = ArrayList<HandlerType>()
+
+        enumValues<HandlerType>().forEach {
```

Comment:
This will get all the handlers including `before`, `ws` and `after`.
I am not sure that the 405 status should be returned when only `before` or `after` handler is set, as these are not actual HTTP methods.

Reply:
You're absolutely correct, before and after are not applicable. WebSocket technically is, but I'm not sure if it should be included either.